flink-core/src/main/java/org/apache/flink/configuration/CheckpointingOptions.java

```diff
@@ -102,13 +102,15 @@
 
     /** Whether to enable state change log. */
     @Documentation.Section(value = Documentation.Sections.COMMON_STATE_BACKENDS)
-    @Documentation.ExcludeFromDocumentation("Hidden for now")
     public static final ConfigOption<Boolean> ENABLE_STATE_CHANGE_LOG =
             ConfigOptions.key("state.backend.changelog.enabled")
                     .booleanType()
-                    .defaultValue(false)
+                    .noDefaultValue()
```

Comment:
As discussed offline, this configuration should have a default value of `false`. We might have different kinds of implementation but we should tell user explicitly that this option is disabled by default.

Reply:
Yes, the default value here is false, but the return value of ```StreamExecutionEnvironment::isChangelogStateBackendEnabled``` is a ```TernaryBoolean```, which may bring some confusion. The ```TernaryBoolean``` is essential cause table planner uses the value to create StreamGraphGenerator (see [here](https://github.com/apache/flink/pull/16153/commits/2c78381d58d1e7e3ee3edd06afb03e7e6eb1f177#diff-c2ed4f07637221036620a06234633cbed96eedf4f29297fd3d766b3e1e751cbaR48)). @rkhachatryan and @curcur would you please to take a look?